### PR TITLE
ucg: rewrite tests

### PR DESCRIPTION
**Pull request by AndersonTorres**
The tests can be easily done in the local machine, there is no necessity of
starting a virtual machine for them.

###### Motivation for this change

sprint https://github.com/NixOS/nixpkgs/issues/143681

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by AndersonTorres** [comment]
It does not build on ARM because of lack of `cpuid.h`.

**Comment by iAmMrinal0** [comment]
I think the relevant entry for `ucg` from `nixos/tests/all-tests.nix` can also be removed.

File: pkgs/tools/text/ucg/default.nix [hunk 1]
```diff
@@ -22,6 +19,20 @@ in stdenv.mkDerivation {
   nativeBuildInputs = [ autoreconfHook pkg-config ];
   buildInputs = [ pcre ];
 
+  installCheckPhase = ''
```
Comment (r-burns) [inline pkgs/tools/text/ucg/default.nix:22]: Be sure to specify doInstallCheck = true so this runs

File: pkgs/tools/text/ucg/default.nix [hunk 2]
```diff
@@ -22,6 +19,20 @@ in stdenv.mkDerivation {
...
+    runHook preInstallCheck
+
+    testFile=$(mktemp /tmp/ucg-test.XXXX)
+    echo -ne 'Lorem ipsum dolor sit amet\n2.7182818284590' > $testFile
+    $out/bin/ucg 'dolor' $testFile && echo OK || echo fail
```
Comment (r-burns) [inline pkgs/tools/text/ucg/default.nix:27]: If any of these commands return nonzero, they will take the `||` path and return the result of the `echo fail` (success) and proceed happily. So we need to either remove the `&&`/`||`, or add another mechanism to make the exit code nonzero.

Comment (AndersonTorres) [reply pkgs/tools/text/ucg/default.nix:27]: Something like `exit -1`?

Comment (iAmMrinal0) [reply pkgs/tools/text/ucg/default.nix:27]: according to the help section of `ucg`:
```
Exit status is 0 if any matches were found, 1 if no matches, 2 or greater on error.
```
so the following should suffice without the echo parts:
```
$out/bin/ucg pattern $testFile
```

**Review by SuperSandro2000 — COMMENTED** [review]
Test fails with ``/nix/store/qcq1y0nfxv8za7w6c682s93gk87r2xy1-stdenv-linux/setup: eval: line 1356: syntax error: unexpected end of file``.

**Comment by AndersonTorres** [comment]
Now they are running OK here.

**Pull request merged**